### PR TITLE
feat: sonarqube account e2e

### PR DESCRIPTION
**Pull request by abeizn**
### Summary
feat: sonarqube account e2e

### Does this close any open issues?
part of #2305 

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_accounts.csv [hunk 1]
```diff
@@ -0,0 +1,6 @@
...
+4,"{""connectionId"":1,""ProjectKey"":""""}","{""login"":""admin3"",""name"":""Administrator3"",""active"":false,""groups"":[""sonar-administrators"",""sonar-users""],""tokensCount"":3,""local"":true,""externalIdentity"":""admin"",""externalProvider"":""sonarqube"",""lastConnectionDate"":""2023-02-09T11:28:44+0000""}","http://35.89.154.221:30008/api/users/search?p=1&ps=100","null","2023-02-09 11:55:28.770"
+5,"{""connectionId"":1,""ProjectKey"":""""}","{""login"":""admin4"",""name"":""Administrator4"",""active"":true,""groups"":[""sonar-administrators"",""sonar-users""],""tokensCount"":3,""local"":true,""externalIdentity"":""admin"",""externalProvider"":""sonarqube"",""lastConnectionDate"":""2023-02-09T11:28:44+0000""}","http://35.89.154.221:30008/api/users/search?p=1&ps=100","null","2023-02-09 11:55:28.770"
```
Comment (warren830) [inline backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_accounts.csv:6]: please add some interfered data:
like connectionId = 2, 3

File: backend/plugins/sonarqube/e2e/snapshot_tables/_tool_sonarqube_accounts.csv [hunk 2]
```diff
@@ -0,0 +1,5 @@
+connection_id,login,batch_id,name,email,active,local
```
Comment (warren830) [inline backend/plugins/sonarqube/e2e/snapshot_tables/_tool_sonarqube_accounts.csv:1]: there should not be batch_id

File: backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_accounts.csv [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+6,"{""connectionId"":2,""ProjectKey"":""xxx""}","{""login"":""admin4"",""name"":""Administrator4"",""active"":true,""groups"":[""sonar-administrators"",""sonar-users""],""tokensCount"":3,""local"":true,""externalIdentity"":""admin"",""externalProvider"":""sonarqube"",""lastConnectionDate"":""2023-02-09T11:28:44+0000""}","http://35.89.154.221:30008/api/users/search?p=1&ps=100","null","2023-02-09 11:55:28.770"
+7,"{""connectionId"":3,""ProjectKey"":""""}","{""login"":""admin4"",""name"":""Administrator4"",""active"":true,""groups"":[""sonar-administrators"",""sonar-users""],""tokensCount"":3,""local"":true,""externalIdentity"":""admin"",""externalProvider"":""sonarqube"",""lastConnectionDate"":""2023-02-09T11:28:44+0000""}","http://35.89.154.221:30008/api/users/search?p=1&ps=100","null","2023-02-09 11:55:28.770"
```
Comment (warren830) [inline backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_accounts.csv:8]: please add an empty line at the end of file

File: backend/plugins/sonarqube/tasks/projects_extractor.go [hunk 4]
```diff
@@ -28,13 +28,39 @@ import (
...
+			body := &models.SonarqubeProject{
+				ConnectionId:     data.Options.ConnectionId,
+				ProjectKey:       res.ProjectKey,
+				Name:             res.Name,
+				Qualifier:        res.Qualifier,
+				Visibility:       res.Visibility,
+				LastAnalysisDate: res.LastAnalysisDate,
+				Revision:         res.Revision,
+			}
+			data.LastAnalysisDate = body.LastAnalysisDate.ToNullableTime()
```
Comment (warren830) [inline backend/plugins/sonarqube/tasks/projects_extractor.go:63]: What is this？ I remember I already changed this

File: backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_projects.csv [hunk 5]
```diff
@@ -0,0 +1,12 @@
+"id","params","data","url","input","created_at"
+642,"{""connectionId"":1,""ProjectKey"":""fa2cf9cd-c448-4fc3-99a5-1c893f15d84c""}","{""key"":""fa2cf9cd-c448-4fc3-99a5-1c893f15d84c"",""name"":""activeadmin"",""qualifier"":""TRK"",""visibility"":""public"",""lastAnalysisDate"":""2022-12-23T13:52:39+0000"",""revision"":""51689e48ec3faa51127b8462b2219e5e0c09ad8a""}","http://35.89.154.221:30008/api/projects/search?p=1&ps=100","null","2023-02-10 02:46:14.044"
```
Comment (warren830) [inline backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_projects.csv:2]: projectKey is empty in param

Comment (warren830) [reply backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_projects.csv:2]: next line

File: backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_projects.csv [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+643,"{""connectionId"":1,""ProjectKey"":""""}","{""key"":""a67b8166-5768-46a5-8830-6d86d797419a"",""name"":""actix-web"",""qualifier"":""TRK"",""visibility"":""public"",""lastAnalysisDate"":""2023-01-04T22:36:46+0000"",""revision"":""08c2cdf6416287d63387584213ea3dd41b007ef6""}","http://35.89.154.221:30008/api/projects/search?p=1&ps=100","null","2023-02-10 02:46:14.044"
+644,"{""connectionId"":2,""ProjectKey"":""""}","{""key"":""a7c268ed-525b-438d-8729-18e83229e66c"",""name"":""Aerial"",""qualifier"":""TRK"",""visibility"":""public"",""lastAnalysisDate"":""2022-12-22T18:26:29+0000"",""revision"":""405bc2494af895f8b9ccbf12f3d0c9530a154011""}","http://35.89.154.221:30008/api/projects/search?p=1&ps=100","null","2023-02-10 02:46:14.044"
+645,"{""connectionId"":2,""ProjectKey"":""""}","{""key"":""f5a50c63-2e8f-4107-9014-853f6f467757"",""name"":""aerosolve"",""qualifier"":""TRK"",""visibility"":""public"",""lastAnalysisDate"":""2023-01-02T06:58:08+0000"",""revision"":""442e76ebb7cbc7f60b04fbfac30dbf862aaffc67""}","http://35.89.154.221:30008/api/projects/search?p=1&ps=100","null","2023-02-10 02:46:14.044"
```
Comment (warren830) [inline backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_projects.csv:5]: same as above

File: backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_projects.csv (hunk 5 above)
Comment (abeizn) [reply backend/plugins/sonarqube/e2e/raw_tables/_raw_sonarqube_projects.csv:2]: Can be understood as interference data？

**Review by warren830 — APPROVED** [review]
LGTM

**Pull request merged**